store/store.go

```diff
@@ -445,6 +500,17 @@ func (bs *BlockStore) SaveSeenCommit(height int64, seenCommit *types.Commit) err
 	return bs.db.Set(calcSeenCommitKey(height), seenCommitBytes)
 }
 
+func (bs *BlockStore) SaveTxIndex(txHash []byte, txIndex *cmtstore.TxIndex) error {
+	bz, err := proto.Marshal(txIndex)
+	if err != nil {
+		panic(fmt.Errorf("error serializing txIndex: %v", err))
+	}
+	if err := bs.db.Set(calcTxHashKey(txHash), bz); err != nil {
+		panic(err)
+	}
+	return nil
+}
```

Comment:
Ideally we batch this as well

Reply:
But don't worry about the performance too much if we're planning on switching the storage engine in the near future